doc/userdoc/_ext/versions.json

```diff
@@ -0,0 +1,57 @@
+{
+	"python": 
+	{
+		"min": 3.8
+	},
+	"pip": 
+	{
+		"recommended": "20.3.3"
```

Comment:
The versions given are currently not synced to any real requirements. This PR is just for the mechanics in Sphinx to substitute hard-coded numbers with data from a file. The tags 'min', 'max', 'recommended' are just human-only labels for different numbers associated to the same record. Where in my opinion the latter one is not a good choice and can be removed once we start replacing the version numbers in the docs.
We didn't want to overload this technical PR with many-files-changed on the documentation content side, though.

Reply:
OK, I see your point. But could we replace all "recommended" with "min", so that we do not give the wrong impression of actually recommending old versions?

Reply:
fixed by @terhorstd in 2f8a4a2